datafusion/core/src/execution/context/mod.rs

```diff
@@ -849,6 +859,28 @@ impl SessionContext {
         let table_paths = table_paths.to_urls()?;
         let session_config = self.copied_config();
         let listing_options = options.to_listing_options(&session_config);
+
+        let option_extension = listing_options.file_extension.clone();
```

Comment:
I am pretty confused about this implementation. If the goal is to detect files that don't end with the expected extension, can't we just do something like

```rust
        for path in &table_paths {
            if !path.as_str().ends_with(&extension) {
                    return exec_err!(
                        "File '{path}' does not match the expected extension '{extension}'"
                    );
            }
        }
```

Reply:
Yes, you are right. I made this logic too complicated.